js/main.js

```diff
@@ -31,18 +31,30 @@ function changeView(dataView) {
   }
 }
 
+var $h1NewAndEditEntry = document.querySelector('#new-edit-entry');
+
 $formEnterNew.addEventListener('submit', function () {
   event.preventDefault();
-  var formObject = {};
-  formObject.title = $formEnterNew.elements.title.value;
-  formObject.url = $formEnterNew.elements.url.value;
-  formObject.notes = $formEnterNew.elements.notes.value;
-  formObject.entryId = data.nextEntryId;
-  data.nextEntryId++;
-  data.entries.unshift(formObject);
+  if ($h1NewAndEditEntry.textContent === 'Edit Entry') {
+    data.entries[data.entries.length - data.editing.entryId].title = $formEnterNew.elements.title.value;
+    data.entries[data.entries.length - data.editing.entryId].url = $formEnterNew.elements.url.value;
+    data.entries[data.entries.length - data.editing.entryId].notes = $formEnterNew.elements.notes.value;
```

Comment:
You should definitely not be using the `id` like an index. What happens if somebody deletes an entry eventually? This code would no longer work. You should loop over your list of entries, and find the entry with the `id` that matches the entry stored in `data.editing`